src/lib/Web3Function.ts

```diff
@@ -165,6 +166,8 @@ export class Web3Function {
         }
       }
     });
-    return provider;
+    const multiChainProvider = new Web3FunctionMultiChainProvider(providerUrl);
```

Comment:
it looks like we instantiate a `provider` above that we don't use?
also the rpc limit handler won't trigger anymore 

I think the `Web3FunctionMultiChainProvider` should be an `EventEmitter` so that we can do:
`multiChainProvider.on("debug", (data) => { ... `

And internally, it should listen all `debug` events from all providers and `emit` them to bubble up.
something like 
`provider.on('debug', data => this.emit('debug', data))`